[static-runtime] one more attempt to improve crash log readability

tenpercent opened the pull request:
Summary:
* add human readable type and ivalue printout
* fix internal linter warnings

Test Plan:
error message now looks like e.g.
```
E0315 16:27:32.409082 422313 ExceptionTracer.cpp:222] exception stack complete
terminate called after throwing an instance of 'c10::Error'
  what():  List[int] is not a subtype of List[int]; schema arg name: 'split_sizes', ivalue: [1, 1]
```

Differential Revision: D44112297

davidberard98 reviewed (approved):
LGMT! But please address the OSS linter. Unlike the internal linter, the OSS linter is mandatory (they'll revert your diff if you break the lint)

tenpercent committed:
[static-runtime] one more attempt to improve crash log readability (#96903)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/96903

* add human readable type and ivalue printout
* fix internal linter warnings

Test Plan:
error message now looks like e.g.
```
E0315 16:27:32.409082 422313 ExceptionTracer.cpp:222] exception stack complete
terminate called after throwing an instance of 'c10::Error'
  what():  List[int] is not a subtype of List[int]; schema arg name: 'split_sizes', ivalue: [1, 1]
```

Reviewed By: davidberard98

Differential Revision: D44112297

fbshipit-source-id: 4451c4472d3c334ffbabd4ae53c926b7ab2b9e78
